rosbag2_storage_default_plugins/src/rosbag2_storage_default_plugins/sqlite/sqlite_statement_wrapper.cpp

```diff
@@ -51,8 +51,9 @@ SqliteStatementWrapper::~SqliteStatementWrapper()
 std::shared_ptr<SqliteStatementWrapper> SqliteStatementWrapper::execute_and_reset()
 {
   int return_code = sqlite3_step(statement_);
-  if (return_code != SQLITE_OK && return_code != SQLITE_DONE) {
-    throw SqliteException("Error processing SQLite statement.");
+  if (return_code != SQLITE_OK && return_code != SQLITE_DONE && return_code != SQLITE_ROW) {
```

Comment:
nitpick: Maybe for readability, these checks can be excluded in a little inline function `is_valid` or something. So ease up the condition.

Reply:
I agree.